models/ROMS_rutgers/model_mod.f90

```diff
@@ -0,0 +1,1716 @@
+Neta_v   = nc_get_dimension_size(ncid, 'eta_v',   routine)
+
+! Read the vertical dimensions 
+Ns_rho   = nc_get_dimension_size(ncid, 's_rho', routine)
+Ns_w     = nc_get_dimension_size(ncid, 's_w'  , routine)
+
+call nc_close_file(ncid, routine)
+
+Nx = Nxi_rho  ! Setting the nominal value of the 'global' variables
+Ny = Neta_rho ! Setting the nominal value of the 'global' variables
+Nz = Ns_rho   ! Setting the nominal value of the 'global' variables
+
+Nu = Nxi_u
+Nv = Neta_v
+Nw = Ns_w
+
+end subroutine get_grid_dimensions
+
+
+!-----------------------------------------------------------------------
+! Read the actual grid values from the ROMS netcdf file.
+! We might need to adapt this to allow for leapfrog time stepping
+! The assumption is that we're reading from history files 
+! which only have one time-level. 
+
+subroutine get_grid()
+
+integer                     :: ncid
+character(len=*), parameter :: routine = 'get_grid'
+
+real(r8), allocatable       :: zeta(:,:), zeta_lf(:,:,:) 
+real(r8), allocatable       :: mask(:,:)                 ! Land mask: 0 land & 1 water
+real(r8)                    :: Zm = -20000.0_r8          ! a masking factor to account for land
+
+allocate(ULAT(Nu, Ny), ULON(Nu, Ny), UMSK(Nu, Ny), UDEP(Nu, Ny, Nz))
+allocate(VLAT(Nx, Nv), VLON(Nx, Nv), VMSK(Nx, Nv), VDEP(Nx, Nv, Nz))
+allocate(TLAT(Nx, Ny), TLON(Nx, Ny), TMSK(Nx, Ny), TDEP(Nx, Ny, Nz))
+
+allocate(h(Nx,Ny), Cr(Nz), sr(Nz))
+
+! s-coordinates at RHO-points
+!do ik = 1, Nz
+!   sr(ik) = (ik - 0.5_r8 - Nz) / Nz
+!enddo
+! s-coordinates at W-points
+!do ik = 0, Nz
+!   sw(ik+1) = (real(ik) - Nz) / Nz
+!enddo
+
+! Read the vertical information 
+ncid = nc_open_file_readonly(roms_filename, routine)
+
+! Varibles needed to calculate depth given the free surface
+call nc_get_variable(ncid, 'h'         , h , routine)
+call nc_get_variable(ncid, 'hc'        , hc, routine)
+call nc_get_variable(ncid, 'Cs_r'      , Cr, routine)
+call nc_get_variable(ncid, 's_rho'     , sr, routine)
+call nc_get_variable(ncid, 'Vtransform', Vt, routine)
+
+! Check whether physical depth data exist in the restart file
+if (nc_variable_exists(ncid, 'z_rho')) then 
+
+   ! It exists, so grab all of them
+   call nc_get_variable(ncid, 'z_rho', TDEP, routine)
+   call nc_get_variable(ncid, 'z_u'  , UDEP, routine)
+   call nc_get_variable(ncid, 'z_v'  , VDEP, routine)
+   TDEP = -TDEP
+   UDEP = -UDEP
+   VDEP = -VDEP
+else
+   
+   ! Need to compute the physical depth
+   ! Check whether zeta is coming from a history or a restart file
+   ! The 3rd dimension shouldn't exist in a history file
+   allocate(zeta(Nx, Ny))
+
+   if (nc_dimension_exists(ncid, 'three')) then 
+      allocate(zeta_lf(Nx, Ny, 3))
+      zeta = zeta_lf(:, :, 1)
+   else 
+      call nc_get_variable(ncid, 'zeta', zeta, routine)
+   endif 
+
+   ! Find depth at the center of the cells
+   do ix = 1, Nx
+      do iy = 1, Ny 
+         call compute_physical_depth(h(ix, iy), zeta(ix, iy), TDEP(ix, iy, :))
```

Comment:
For memory you probably want to do this on the fly rather than storing a state variable sized array for TDEP, UDEP, VDEP.